### PR TITLE
Let's add superagent to the benchmark. closes #2730

### LINKED ISSUE #2730: Let's add superagent to the benchmark table.
## This would solve...

I'm building a [library](https://github.com/eddienubes/sagetest) similar to **supertest**, but on top of **undici**. As an advantage, I'd like point out that **undici** is faster than **superagent** (which **supertest** uses under the hood).

In my opinion, for this benchmark it makes sense to reside in **undici**'s repo.

By the way, seems like **superagent** doesn't reuse established connections on purpose, so it'll be quite slow.

## The implementation should look like...

[Pull Request](https://github.com/nodejs/undici/pull/2731)

### PR DESCRIPTION
**Pull request by eddienubes**
## This relates to...

Adding superagent to the benchmark 
#2730 

## Rationale

I'm building a [library](https://github.com/eddienubes/sagetest) similar to supertest, but on top of undici. As an advantage, I'd like point out that undici is faster than superagent (which supertest uses under the hood).

In my opinion, for this benchmark it makes sense to reside in undici's repo.

## Changes

- added **superagent** call to benchmark.js
- added **superagent** and **@types/superagent** as devDependencies.

Also, I haven't changed the README yet. I'd like someone from maintainers to run this benchmark to make sure proper arguments were passed (samples, connections etc.)

----

- [x] I have read and agreed to the [Developer's Certificate of Origin][cert]
- [x] Tested
- [x] Benchmarked (**optional**)
- [x] Documented
- [x] Review ready
- [x] In review
- [x] Merge ready

[cert]: https://github.com/nodejs/undici/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: benchmarks/benchmark.js [hunk 1]
```diff
@@ -318,6 +319,16 @@ if (process.env.PORT) {
       }).catch(console.log)
     })
   }
+
+  experiments.superagent = () => {
+    return makeParallelRequests(resolve => {
+      superagent.get(dest.url).pipe(new Writable({
```
Comment (mcollina) [inline benchmarks/benchmark.js:325]: You should be creating and setting and agent, otherwise the benchmark would not be even

**Review by mcollina — CHANGES_REQUESTED** [review]

File: benchmarks/benchmark.js (hunk 1 above)
Comment (eddienubes) [reply benchmarks/benchmark.js:330]: @mcollina 
Superagent doesn't expose a verbose API for setting such agent, so I used a workaround.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by ronag — APPROVED** [review]

**Comment by eddienubes** [comment]
@ronag @mcollina 
I think we didn't update README, or you'd like to do it in a separate PR?

**Comment by ronag** [comment]
It would be have to be in a separate PR now. Someone just needs to run the benchmarks.

**Comment by eddienubes** [comment]
I can handle that.
Do I just run it with defaults set in benchmark.js?

**Comment by ronag** [comment]
> I can handle that. Do I just run it with defaults set in benchmark.js?

I think so.

**Comment by mcollina** [comment]
I'll update the benchmark at the first occasion

**Pull request merged**